Update the docs to the latest API properties

ChaseWiseman opened the pull request:
## Summary

Adds profile field definitions to the User Membership doc pages.

Also adds docker config files as an alternative to running natively.

### Before merging to master

- [x] Memberships 1.19.0 has been released

### After merging to master

- [x] [Deploy the update](https://github.com/skyverge/woocommerce-memberships-rest-api-docs#deploying-an-update)

unfulvio reviewed (changes requested):
hey @ChaseWiseman for the documentation part this looks good.

May I know why you removed the reference to Slate user guide? I thought it could be an helpful resource / reminder for us how this guide was generated.

Also the instructions weren't necessarily for Vagrant, I always run the build locally from my local machine. I didn't think you really needed docker or vagrant to build this. As long as you have Ruby/Bundle you'll be fine. Vagrant has nothing to do with it.

ChaseWiseman commented:
@unfulvio thanks!

>May I know why you removed the reference to Slate user guide? I thought it could be an helpful resource / reminder for us how this guide was generated.

Not sure what you mean - the references [still exist here](https://github.com/skyverge/woocommerce-memberships-rest-api-docs/pull/6/files#diff-04c6e90faac2675aa89e2176d2eec7d8R54-R56)

>Also the instructions weren't necessarily for Vagrant, I always run the build locally from my local machine. I didn't think you really needed docker or vagrant to build this. As long as you have Ruby/Bundle you'll be fine. Vagrant has nothing to do with it.

Yeah I meant to remove the vagrant reference, my bad. Corrected with 8ad5bc2f455e9ccba67daab0c8c53b3b3bf19084

I added docker because it's a nice isolated option vs. native. I had Ruby dependency errors when trying to install from the start, and I've dealt with situations where correcting those errors ends up corrupting other projects' dependencies, which is not worth it just to get this small package running. I figured I'd add that alternative to the readme for other folks as well.

unfulvio commented:
thanks for tweaking the readme, @ChaseWiseman sorry I was wrong thought the Slate reference was gone, I was wrong

I am approving but not merging this yet perhaps, since Memberships 1.19 wasn't deployed yet - I'll make a note in the epic on CH to remember to merge this && run a  build for deploying it
